src/libudev/libudev-list-internal.h

```diff
@@ -3,22 +3,17 @@
 
 #include "libudev.h"
 
-struct udev_list_node {
-        struct udev_list_node *next, *prev;
-};
+#include "hashmap.h"
+#include "list.h"
 
 struct udev_list {
-        struct udev *udev;
-        struct udev_list_node node;
-        struct udev_list_entry **entries;
-        unsigned entries_cur;
-        unsigned entries_max;
+        Hashmap *hashmap;
         bool unique;
+
+        LIST_HEAD(struct udev_list_entry, entries);
 };
```

Comment:
IIUC the list here is used only as support storage in `udev_hwdb_get_properties_list_entry()` to be able to return an  ordered list. Wouldn't it be better to use an `OrderedHashmap` instead? The implementation currently in the codebase only allows to store elements in insertion order, but it should not be hard to add a `sort` method to it.

Reply:
If I understand correctly, libudev-list sorts entries with its name, and optionally drops duplicated entries which have same name. So, I re-implement this by using 'Hashmap' and 'qsort'. After sorting, list structure is initialized.

Note that `OrderedHashmap` does not sort entries. It keeps the insertion order.